extensions/deployers/licenses.py

```diff
@@ -1,42 +1,29 @@
 import os
 import zipfile
 from conan.api.output import ConanOutput
-from conan.tools.files import copy, rmdir
+
 
 # USE **KWARGS to be robust against changes
 def deploy(graph, output_folder, **kwargs):
     out = ConanOutput("deployer(licenses)")
     conanfile = graph.root.conanfile
-    files = []
-    
-    # Cleanup before we start :)
-    tmp_dir = os.path.join(output_folder, "licenses")
-    if os.path.exists(tmp_dir):
-        rmdir(conanfile, tmp_dir)
 
     # For each dep
-    for r, d in conanfile.dependencies.items():
-        if d.package_folder is None:
-            continue
-        
-        # Look for a licenses folder
-        search_dir = os.path.join(d.package_folder, "licenses") # This is the CCI convention
-        if not os.path.isdir(search_dir):
-            continue
+    with zipfile.ZipFile(os.path.join(output_folder, 'licenses.zip'), 'w') as licenses_zip:
```

Comment:
I like somehow that the deployer has separated the recollection and the zipping, so users that don't want the final ``zip``, they can just remove those lines from their fork of this deployer.

But I think it is also good as this proposes, not a big deal, clear enough ,and it handles the nested folder, so already an improvement over the existing one, good for me.